mbedtls/src/ssl/context.rs

```diff
@@ -99,19 +110,22 @@ impl Context {
         };
 
         Context {
-            inner,
+            inner: HandshakeContext {
+                inner,
+                handshake_ca_cert: None,
+                handshake_crl: None,
+                
+                handshake_cert: vec![],
+                handshake_pk: vec![],
+            },
             config: config.clone(),
             io: None,
-
-            handshake_ca_cert: None,
-            handshake_crl: None,
-            
-            handshake_cert: vec![],
-            handshake_pk: vec![],
         }
     }
+}
 
-    pub fn establish<T: IoCallback + Send + Sync + 'static>(&mut self, io: T, hostname: Option<&str>) -> Result<()> {
+impl<T: IoCallback + Send + Sync + 'static> Context<T> {
```

Comment:
I thought part of the point was to get rid of the `Send + Sync + 'static` requirement here?

Reply:
This is needed for that work and async.

I will raise another PR with that part.